pyk/src/pyk/ktool/claim_loader.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import annotations
+
+import json
+from typing import TYPE_CHECKING, NamedTuple
+
+from ..kast.outer import KFlatModuleList
+from ..kast.utils import slurp_definitions
+from ..utils import hash_str
+from .claim_index import ClaimIndex
+
+if TYPE_CHECKING:
+    from collections.abc import Iterable, Mapping
+    from pathlib import Path
+    from typing import Any
+
+    from ..kast.outer import KClaim
+    from . import TypeInferenceMode
+    from .kprove import KProve
+
+
+class ClaimLoader:
+    _kprove: KProve
+
+    def __init__(self, kprove: KProve):
+        self._kprove = kprove
+
+    def load_claims(
+        self,
+        spec_file: Path,
+        *,
+        spec_module_name: str | None = None,
+        include_dirs: Iterable[Path] = (),
+        md_selector: str | None = None,
+        claim_labels: Iterable[str] | None = None,
+        exclude_claim_labels: Iterable[str] | None = None,
+        include_dependencies: bool = True,
+        type_inference_mode: TypeInferenceMode | None = None,
+    ) -> list[KClaim]:
+        digest = self._digest(spec_file, include_dirs=include_dirs, md_selector=md_selector)
+
+        claim_file = spec_file.with_suffix('.json')
+
+        cache_hit = False
+        if claim_file.exists():
+            module_list, _digest = _ClaimModuleList.from_dict(json.loads(claim_file.read_text()))
+            cache_hit = digest == _digest
```

Comment:
Maybe call this `loaded_digest` or something? It caught me out very briefly scanning from the top that there's `self._digest` as well

Reply:
Done.